gpytorch/kernels/arc_kernel.py

```diff
@@ -0,0 +1,181 @@
+"""
+Created on Tue Jan  7 11:13:37 2020
+
+@author: kostal
+"""
+
+from gpytorch.constraints import Positive
+from gpytorch.kernels import Kernel, MaternKernel
+import torch
+from math import pi
+from typing import Optional
+from gpytorch.priors import Prior
+
+
+class ArcKernel(Kernel):
+
+    r""" Computes a covariance matrix based on the Arc Kernel
+    (https://arxiv.org/abs/1409.4011) between inputs :math:`\mathbf{x_1}`
+    and :math:`\mathbf{x_2}`. First it applies a cylindrical embedding:
+
+    .. math::
+
+
+          g_{i}(\mathbf{x}) = \{\begin{eqnarray}
+          [0, 0]^{T} \qquad if\;\delta_{i}(\mathbf{x}) = false\\
+          \omega_{i}[\sin{\pi\rho_{i}\frac{x_{i}}{u_{i}-l_{i}}}, \cos{\pi\rho_{i}\frac{x_{i}}{u_{i}-l_{i}}}] \qquad otherwise
+          \end{eqnarray}
+
+
+    where
+
+    * :math:`\rho` is the angle parameter.
+    * :math:`\omega` is a radius parameter.
+
+    then the kernel is built with the particular covariance function, e.g.
+
+    .. math::
+        \begin{equation}
+        k_{i}(\mathbf{x}, \mathbf{x^{'}}) = \sigma^{2}\exp(-\frac{1}{2}d_{i}(\mathbf{x}, \mathbf{x^{'}}))^{2}
+        \end{equation}
+
+    and the produt between dimensions
+
+    .. math::
+        \begin{equation}
+        k_{i}(\mathbf{x}, \mathbf{x^{'}}) = \sigma^{2}\exp(-\frac{1}{2}d_{i}(\mathbf{x}, \mathbf{x^{'}}))^{2}
+        \end{equation}
+    .. note::
+
+        This kernel does not have an `outputscale` parameter. To add a scaling
+        parameter, decorate this kernel with a
+        :class:`gpytorch.kernels.ScaleKernel`.
+        When using with an input of `b x n x d` dimensions, decorate this
+        kernel with :class:`gpytorch.kernel.ProductStructuredKernel , setting
+        the number of dims, `num_dims to d.`
+    .. note::
+
+        This kernel does not have an ARD lengthscale option.
+
+    Args:
+        :attr:`base_kernel` (gpytorch.kernels.Kernel):
+            The euclidean covariance of choice. Default: `MaternKernel(nu=2.5)`
+        :attr:`ard_num_dims` (int):
+            The number of dimensions to compute the kernel for. The kernel has
+            two parameters which are individually defined for each dimension.
+            Default: `None`.
+        :attr:`angle_prior` (Prior, optional):
+            Set this if you want to apply a prior to the period angle
+            parameter.  Default: `None`.
+        :attr:`radius_prior` (Prior, optional):
+            Set this if you want to apply a prior to the lengthscale parameter.
+            Default: `None`.
+
+    Attributes:
+        :attr:`radius` (Tensor):
+            The radius parameter. Size = `*batch_shape  x 1`.
+        :attr:`angle` (Tensor):
+            The period angle parameter. Size = `*batch_shape  x 1`.
+
+    Example:
+        >>> x = torch.randn(10, 5)
+        >>> # Non-batch: Simple option
+        ... base_kernel = gpytorch.kernels.MaternKernel(nu=2.5)
+        >>> base_kernel.raw_lengthscale.requires_grad_(False)
```

Comment:
It occurs to me just now that we could also turn off the base kernel's lengthscale in the constructor for ArcKernel. Something like: check base_kernel.has_lengthscale, if True set it to 1 and set requires grad to false as here.

What do you think?

Reply:
I do agree, let me change the code and the docs

Reply:
An error occurs:

```
RuntimeError: you can only change requires_grad flags of leaf variables. If you want to use a computed variable in a subgraph that doesn't require differentiation use var_no_grad = var.detach()
```

Reply:
Added it in the initialization of the kernel